site/testsuites/testsuite-javascript/__tests__/tbdex/pfi/quotes.js

```diff
@@ -0,0 +1,285 @@
+  },
+
+  getRfq: (opts) => {
+    // Mock data for getRfq
+    const mockData = { id: opts.exchangeId, type: 'Rfq' };
+    return Promise.resolve(mockData);
+  },
+
+  getQuote: (opts) => {
+    // Mock data for getQuote
+    const mockData = { id: opts.exchangeId, type: 'Quote' };
+    return Promise.resolve(mockData);
+  },
+
+  getOrder: (opts) => {
+    // Mock data for getOrder
+    const mockData = { id: opts.exchangeId, type: 'Order' };
+    return Promise.resolve(mockData);
+  },
+
+  getOrderStatuses: (opts) => {
+    // Mock data for getOrderStatuses
+    const mockData = [{ status: 'Pending' }, { status: 'Completed' }];
+    return Promise.resolve(mockData);
+  },
+
+  getClose: (opts) => {
+    // Mock data for getClose
+    const mockData = { id: opts.exchangeId, type: 'Close' };
+    return Promise.resolve(mockData);
+  },
+
+
+  async write({ message }) {
+      await dataProvider.insert('exchange', {
+              exchangeid: message.exchangeId,
+              messagekind: message.kind,
+              messageid: message.id,
+              subject: message.subject,
+              message: JSON.stringify(message)
+          });
+  }  
+
+};
+
+const offeringsApiProvider = {
+  getOffering: (opts) => {
+      dataProvider.get('offering', opts.id).then(([result]) => {
+          if (!result) {
+              return undefined
+          }
+          return Offering.factory(result.offering)
+      });
+  },
+
+  getOfferings: (opts) => {
+    dataProvider.query('offering', "*").then((results) => {
+        const offerings = []
+
+        for (let result of results) {
+            const offering = Offering.factory(result.offering)
+            offerings.push(offering)
+        }
+
+        return offerings
+    });
+  },
+
+
+  async create(offering) {
+      await dataProvider.insert('offering', {
+          offeringid: offering.id,
+          payoutcurrency: offering.payoutCurrency.currencyCode,
+          payincurrency: offering.payinCurrency.currencyCode,
+          offering: JSON.stringify(offering)
+      });
+  }
+  
+};  
+
+//---------------------------------------------------------------------------//
+//---------------------------------------------------------------------------//
+
+async function createQuoteFromRfq(message) {
+    // :snippet-start: pfiQuotesWriteJs
+    // Write the message to your exchanges database
+    await dataProvider.insert('exchange', {
+        exchangeid: message.exchangeId,
+        messagekind: message.kind,
+        messageid: message.id,
+        subject: message.subject,
+        message: JSON.stringify(message)
+    });
+
+    //highlight-start
+    const offering = await offeringsApiProvider.getOffering(message.offeringId)
+    //highlight-end
+    // :snippet-end:
+
+    // :snippet-start: pfiQuotesProcessJs
+    const rfqOptions = {
+        data: message.data(), 
+        metadata: message.metadata()
+    };
+
+    const rfq = Rfq.create(rfqOptions)
+
+    try {
```

Comment:
```suggestion
    // :snippet-start: pfiQuotesProcessJs
    try {
```

The PFI receives the RFQ, they don't create it. so showing a snippet where they create a RFQ with no requirements and then verifying the requirements 2 lines later doesn't make sense